File: API/ICCMS-API/Models/Project.cs
using Google.Cloud.Firestore;

namespace ICCMS_API.Models
{
    [FirestoreData]
    public class Project
    {
        [FirestoreProperty("projectId")]
        public string ProjectId { get; set; } = string.Empty;

        [FirestoreProperty("projectManagerId")]
        public string ProjectManagerId { get; set; } = string.Empty;

        [FirestoreProperty("clientId")]
        public string ClientId { get; set; } = string.Empty;

        [FirestoreProperty("name")]
        public string Name { get; set; } = string.Empty;

        [FirestoreProperty("description")]
        public string Description { get; set; } = string.Empty;

        [FirestoreProperty("budgetPlanned")]
        public double BudgetPlanned { get; set; }

        [FirestoreProperty("budgetActual")]
        public double BudgetActual { get; set; }

        [FirestoreProperty("status")]
        public string Status { get; set; } = string.Empty;

        [FirestoreProperty("isDraft")]
        public bool IsDraft { get; set; } = false;

        [FirestoreProperty("startDatePlanned")]
        public DateTime StartDate { get; set; }

        [FirestoreProperty("endDatePlanned")]
        public DateTime EndDatePlanned { get; set; }

        [FirestoreProperty("endDateActual")]
        public DateTime? EndDateActual { get; set; }

        // Draft/ownership/housekeeping fields
        [FirestoreProperty("createdByUserId")]
        public string CreatedByUserId { get; set; } = string.Empty;

        [FirestoreProperty("updatedAt")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: Mobile/app/src/main/java/com/example/iccms_mobile/data/models/Project.kt
package com.example.iccms_mobile.data.models

import com.google.gson.annotations.SerializedName

data class Project(
    @SerializedName("ProjectId")
    val ProjectId: String = "",
    
    @SerializedName("ProjectManagerId")
    val ProjectManagerId: String = "",
    
    @SerializedName("ClientId")
    val ClientId: String = "",
    
    @SerializedName("Name")
    val Name: String = "",
    
    @SerializedName("Description")
    val Description: String = "",
    
    @SerializedName("BudgetPlanned")
    val BudgetPlanned: Double = 0.0,
    
    @SerializedName("BudgetActual")
    val BudgetActual: Double = 0.0,
    
    @SerializedName("Status")
    val Status: String = "",
    
    @SerializedName("StartDatePlanned")
    val StartDatePlanned: String = "",
    
    @SerializedName("EndDatePlanned")
    val EndDatePlanned: String = "",
    
    @SerializedName("EndDateActual")
    val EndDateActual: String? = null
)
